File: MediCareWebSpa/src/app/DTOs/models/patient.ts
import { Appointment } from "./appointment";
import { Feedback } from "./feedback";
import { User } from "./user";

export class Patient {
  userId: number;
  user: User;
  registerDate: Date;
  birthDate: Date;
  patientCard: string;
  appointments: Appointment[] = [];
  feedbacks: Feedback[] = [];

  constructor(
    userId: number,
    user: User,
    registerDate: Date,
    birthDate: Date,
    patientCard: string
  ) {
    this.userId = userId;
    this.user = user;
    this.registerDate = registerDate;
    this.birthDate = birthDate;
    this.patientCard = patientCard;
  }
}

File: MediCareWebApi/Models/Patient.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MediCare.Models
{
	public class Patient
	{
		[Key]
		public int UserId { get; set; }

		[ForeignKey("UserId")]
		public User User { get; set; }

		[Required]
		public DateTime RegisterDate { get; set; }

		[Required]
		public DateTime BirthDate { get; set; }

		[MaxLength(20)]
		public string? PatientCard { get; set; }

		public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
		public ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();
	}
}
